flict/flictlib/project/reader.py

```diff
@@ -170,6 +183,22 @@ def _read_spdx_2_2(self, only_packages=None):
                     _pkg_name = _pkg['id']
                     if dep_package_name == _pkg_name:
                         packages[top_package]['dependencies'].append({_pkg_name: _pkg})
+            else:
+                dep_id = dep['spdxElementId']
+                for package in self.project['packages']:
+                    if dep_id == package['SPDXID']:
+                        if top_package not in packages:
+                            logging.warning(f"package {top_package} marked as a relationship, but without corresponding package definition.")
+                        else:
+                            _pkg = {
+                                'id': dep_id,
+                                'name': pkg.get('name', ""),
```

Comment:
Again here the use of `get` might lead to more problems than it solves

Reply:
fixed in coming push